File: AZ-204_Develop_Solutions_That_Use_Blob_Storage/code/interact.cs
using Azure.Storage.Blobs;
using Azure.Storage.Blobs.Models;
using System;
using System.IO;
using System.Threading.Tasks;

namespace BlobStorageInteract
{
    class Program
    {
        static async Task Main(string[] args)
        {
            string connectionString = "<your_connection_string>";
            string containerName = "<your_container_name>";
            string blobName = "<your_blob_name>";

            // Create a BlobServiceClient object using the connection string
            BlobServiceClient blobServiceClient = new BlobServiceClient(connectionString);

            // Create a container client object
            BlobContainerClient containerClient = blobServiceClient.GetBlobContainerClient(containerName);

            // Create a blob client object
            BlobClient blobClient = containerClient.GetBlobClient(blobName);

            // Upload a file to the blob storage
            await using FileStream uploadFileStream = File.OpenRead("<path_to_local_file>");
            await blobClient.UploadAsync(uploadFileStream, true);

            // Download the blob to a local file
            BlobDownloadInfo download = await blobClient.DownloadAsync();
            await using FileStream downloadFileStream = File.OpenWrite("<path_to_save_file>");
            await download.Content.CopyToAsync(downloadFileStream);
            downloadFileStream.Close();

            // Delete the blob
            await blobClient.DeleteAsync();

            Console.WriteLine("Blob storage interaction completed.");
        }
    }
}

File: AZ-204_Develop_Solutions_That_Use_Blob_Storage/code/interact.py
from azure.storage.blob import BlobServiceClient

# Connect to the storage account
connection_string = "<your_connection_string>"
blob_service_client = BlobServiceClient.from_connection_string(connection_string)

# Create a container
container_name = "<your_container_name>"
container_client = blob_service_client.create_container(container_name)

# Upload a blob to the container
blob_name = "<your_blob_name>"
blob_client = container_client.get_blob_client(blob_name)
with open("<path_to_local_file>", "rb") as data:
    blob_client.upload_blob(data)

# Download a blob from the container
downloaded_blob = blob_client.download_blob()
with open("<path_to_save_file>", "wb") as file:
    file.write(downloaded_blob.readall())

# List all blobs in the container
blobs = container_client.list_blobs()
for blob in blobs:
    print(blob.name)

# Delete a blob
blob_client.delete_blob()

# Delete the container
container_client.delete_container()